### PR TITLE
fixing notification handlers

### PR DESCRIPTION
**Pull request by someara**
Hello!

This PR will make chef-client go a lottapercent faster on hosts with multiple checks that need to be restarted.

-s

### CODE REVIEW

File: providers/monitor.rb [hunk 1]
```diff
@@ -25,11 +25,6 @@ def whyrun_supported?
     )
     cookbook new_resource.cookbook
     sensitive true if Chef::Resource.instance_methods(false).include?(:sensitive)
-    notifies :restart, 'service[datadog-agent]', :delayed if node['datadog']['agent_start']
```
Comment (olivielpeau) [inline providers/monitor.rb]: Any reason for removing the condition `if node['datadog']['agent_start']` from the `notifies` properties that you added in the recipes?

**Comment by olivielpeau** [comment]
Thanks @someara!

I've added a comment on the PR, let me know what you think. I'll see if we can include your change in the next minor release.

(For reference, the present PR changes a behavior introduced by #229, see discussion on that PR)

**Comment by someara** [comment]
Nice catch. I'll update the PR!

**Comment by someara** [comment]
Done!

**Comment by olivielpeau** [comment]
Thanks @someara, looks good!

Note to self: Once this is merged we need to make sure that all the new recipes use `datadog_monitor` with `notifies`

**Comment by olivielpeau** [comment]
@someara: Actually, after giving this some thought I don't think we can include this in a minor release: the `datadog_monitor` resource is used by some users in their own recipes, and this PR changes the resource's behavior (the resource doesn't take care of restarting the Agent anymore, so existing uses of the resource wouldn't behave as expected).

For this reason I'm slating this PR for our next major release. Let me know what you think, thanks!

**Comment by remeh** [comment]
Hey @someara thanks for your contribution.
I'll work on resolving the conflicts of this pull-request in order to merge it to master soon. Could you please select the "Allow edits from maintainers" option, otherwise I'll have to reproduce the changes in another branch and you'll lose the attribution in the git history 😞

**Comment by remeh** [comment]
I have created https://github.com/DataDog/chef-datadog/pull/596 with your changes (cherry-picking your commits) and solved the merge conflicts. I'm closing this PR, in order to merge the other one, thanks again 👍

**Pull request closed without merging**